### PR TITLE
fix: skip pending payment check on void

### PR DESCRIPTION
**Pull request by dni**
skips payment check if voidwallet is active

### CODE REVIEW

**Comment by motorina0** [comment]
why?

**Comment by dni** [comment]
if voidwallet is active or enabled, payments should not be checked

File: lnbits/tasks.py [hunk 1]
```diff
@@ -133,8 +133,14 @@ async def check_pending_payments():
...
     checked only once, outgoing pending payments will be checked regularly.
     """
+    sleep_time = 60 * 30  # 30 minutes
 
     while settings.lnbits_running:
+        funding_source = get_funding_source()
+        if funding_source.__class__.__name__ == "VoidWallet":
+            logger.warning("Task: skipping pending check for VoidWallet")
+            await asyncio.sleep(sleep_time)
+            continue
```
Comment (motorina0) [inline lnbits/tasks.py:143]: Why not exit the loop and let the task complete. There is no way to switch from `VoidWallet` to a real wallet.
```suggestion
            return
```

File: lnbits/tasks.py [hunk 2]
```diff
@@ -133,8 +133,14 @@ async def check_pending_payments():
     the backend and also to delete expired invoices. Incoming payments will be
     checked only once, outgoing pending payments will be checked regularly.
     """
+    sleep_time = 60 * 30  # 30 minutes
 
     while settings.lnbits_running:
+        funding_source = get_funding_source()
+        if funding_source.__class__.__name__ == "VoidWallet":
```
Comment (motorina0) [inline lnbits/tasks.py:140]: As a side note:
 - maybe funding sources should have an `info()` method (defined at the `base.Wallet` level as abstract)
 - this method would return info about the funding source supported features. Eg:
```json
{
    "invoices_stream": true, // some funding sources do not implement `paid_invoices_stream()` 
    "real_backend": true // in this case only VoidWallet would have false
}
```

**Review by talvasconcelos — APPROVED** [review]

**Review by motorina0 — APPROVED** [review]

**Pull request merged**